File: TextEditor/TextEditor/EditorMemento.cs
namespace TextEditor;

public class EditorMemento
{
    public string Content { get; private set; }
    public int CursorPosition { get; private set; }
    public List<string> Selections { get; private set; }

    public EditorMemento(string content, int cursorPosition, List<string> selections)
    {
        Content = content;
        CursorPosition = cursorPosition;
        Selections = selections;
    }
}

File: TextEditor/src/textEditor/EditorMemento.java
package textEditor;

public class EditorMemento {
    private String content;
    private int cursorPosition;
    private String selection;

    public EditorMemento(String _content, int _cursorPosition, String _selection){
        this.content = _content;
        this.cursorPosition = _cursorPosition;
        this.selection= _selection;

    }

    public String getContent(){
        return content;
    }

    public int getCursorPosition(){
        return cursorPosition;
    }

    public String getSelection(){
        return selection;
    }
}
